sw/host/tests/manuf/provisioning/src/main.rs

```diff
@@ -59,20 +72,71 @@ fn provisioning(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
     }
     device_pk_sec1_bytes.push(0x04); // This indicates the EC public key is not compressed.
     device_pk_sec1_bytes.reverse();
-    let _device_pk = PublicKey::<NistP256>::from_sec1_bytes(&device_pk_sec1_bytes)?;
+    let device_pk = PublicKey::<NistP256>::from_sec1_bytes(&device_pk_sec1_bytes)?;
+
+    // Peform ECDH to generate shared secret (AES) key.
+    let ecdh_shared_secret = diffie_hellman(hsm_sk.to_nonzero_scalar(), device_pk.as_affine());
+    let mut aes_key_bytes = Vec::from(ecdh_shared_secret.raw_secret_bytes().as_slice());
+    aes_key_bytes.reverse();
```

Comment:
Do you have any details about why this reverse is needed? Wondering if I've messed up the endianness somewhere in the AES/ECDH cryptolib API or if this mismatch is between the rust crates.